lib/index-hints.ts

```diff
@@ -1,14 +1,13 @@
-'use strict';
-
 /**
  * An enum of index hints to be used in mysql for querying with index hints
  *
  * @property USE
  * @property FORCE
  * @property IGNORE
  */
-const IndexHints = module.exports = { // eslint-disable-line
-  USE: 'USE',
-  FORCE: 'FORCE',
-  IGNORE: 'IGNORE'
-};
+// eslint-disable-next-line no-shadow
```

Comment:
What are we shadowing?

Reply:
I have no idea! I think it's no-shadow being broken with typescript. Should be fixed once we install `@typescript/no-shadow` instead

Reply:
Maybe just remove it?

Reply:
I'd prefer to keep it for now. I'll try to see about writing a better eslint config & fix all issues in that PR :)

Speaking of which [I have a preset for eslint](https://github.com/ephys/eslint-config) that I've developed over the years initially based on airbnb. Could be a good starting point?

Reply:
I'm fine with any config really :) as long as my editor understands how to format the file for me.

Reply:
I think we should format the entire code base as a one off thing once we settled on a config. Wanna file a pr?

Reply:
Sure :)

Reply:
I believe we wanted to wait until the TypeScript migration was done so we wouldn't have to keep any specific rules for vanilla JS and we could do a proper config at once. But we can try out the preset of @ephys if it doesn't give many issues with the current JS codebase

Reply:
Well, only 12290 errors!

Reply:
After running it with the --fix flag or before?

Reply:
There are some rules I'll disable for the JS codebase, but keep enabled for TS. This should ease the transition.

Reply:
Down to 900 errors. There are a lot of autofixed changes, which will be annoying for existing PRs. It's designed to minimise diff noise in the future so I think it's still worth it.

Reply:
578 errors left but these are actual code smells that should be properly fixed (like using `==`, `new Array`, etc)

Reply:
Hope you'll do that in another pr 😅

Reply:
Yeah :)

Reply:
Shadowing resolved!